### PR TITLE
COM-964: support multiselect in contact attributes

### PR DESCRIPTION
**Pull request by juliawegmayr**
- Add demo contact attribute (branch) allowing to select multiple fields.
- Add demo filter for branch for testing.

Logic for assigning contacts to filters:
- If multiple filter are set in a target group, the contact has to have all those attributes. (e.g. SALUTATION: "Female", BRANCH: "Marketing" -> contact needs both attributes to be assigned.
- If a filter value is an array, one matching  value is required for assigning a contact. (e.g. BRANCH: ["Marketing", "Products"] -> contact only needs one of the values.

Contact Attributes:
Screenshot 2024-09-20 at 09 31 48

Filter:
Screenshot 2024-09-25 at 09 36 38

### CODE REVIEW

File: packages/api/src/target-group/target-groups.service.ts [hunk 1]
```diff
@@ -37,16 +37,23 @@ export class TargetGroupsService {
     ): boolean {
         if (filters && contactAttributes) {
             for (const [key, value] of Object.entries(filters)) {
-                if (!value) continue;
+                if (!value || value.length === 0) continue;
+
+                if (Array.isArray(contactAttributes[key])) {
+                    for (const attribute of contactAttributes[key]) {
+                        if (value.includes(attribute)) {
+                            return true;
```
Comment (RainbowBunchie) [inline packages/api/src/target-group/target-groups.service.ts:45]: You shouldn't return true in this case. Then the other attributes that come after the array will not be checked anymore. 
Example filters (note the order of the filters):
```
BRANCH -> [NEWS] 
SALUTATION -> FEMALE 
```
This would return true for contact since you instantly return true when checking the branch array (which is wrong)
```
BRANCH -> NEWS 
SALUTATION -> MALE
``` 
you should only skip the remaining attributes in the for loop (line 43) in this case.

File: packages/api/src/brevo-contact/brevo-contacts.service.ts [hunk 2]
```diff
@@ -128,10 +128,6 @@ export class BrevoContactsService {
                 if (contactIsInTargetGroupByAttributes) {
                     targetGroupIds.push(targetGroup.brevoId);
                 }
-
-                if (targetGroup.assignedContactsTargetGroupBrevoId && contact?.listIds.includes(targetGroup.assignedContactsTargetGroupBrevoId)) {
-                    targetGroupIds.push(targetGroup.brevoId, targetGroup.assignedContactsTargetGroupBrevoId);
-                }
```
Comment (RainbowBunchie) [inline packages/api/src/brevo-contact/brevo-contacts.service.ts:134]: Why did you remove this? The manually assigned contact groups do not get added anymore if you remove this

**Review by RainbowBunchie — CHANGES_REQUESTED** [review]

**Review by RainbowBunchie — CHANGES_REQUESTED** [review]

File: packages/api/src/brevo-contact/brevo-contact.resolver.ts [hunk 3]
```diff
@@ -143,6 +143,7 @@ export function createBrevoContactResolver({
 
             const updatedNonMainListIds = await this.brevoContactsService.getTargetGroupIdsForExistingContact({
                 contact,
+                scope,
```
[the remark this answers is not in the document]
Comment (juliawegmayr) [reply packages/api/src/brevo-contact/brevo-contact.resolver.ts]: I fixed it again, by removing the scope from the where args as it is not neccessary here

File: packages/api/src/brevo-contact/brevo-contacts.service.ts [hunk 4]
```diff
@@ -112,7 +112,7 @@ export class BrevoContactsService {
         let totalCount = 0;
         const targetGroupIds: number[] = [];
         const limit = 50;
-        const where = { isMainList: false, scope };
+        const where = { isMainList: false };
```
[the remark this answers is not in the document]
Comment (juliawegmayr) [reply packages/api/src/brevo-contact/brevo-contacts.service.ts:109]: Yes, removing is possible. The function is only used in the updateBrevoContact, where the contact is already checked for the scope, so that it's not neccessary here anymore.

File: .changeset/fifty-radios-wonder.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/brevo-api": minor
```
Comment (raphaelblum) [inline .changeset/fifty-radios-wonder.md:2]: I would say this is a patch. I differentiate it like this:

patch: bugfixes, security updates or small changes
minor: New Features like adding test list functionality
major: Breaking Changes, big new Features that greatly change the use of the module and so on

Comment (juliawegmayr) [reply .changeset/fifty-radios-wonder.md]: Thank you for the explanation! :)

**Review by RainbowBunchie — APPROVED** [review]

**Review by raphaelblum — APPROVED** [review]
